### ISSUE TITLE
Prisma CLI installation hangs indefinitely inside of docker (Kernel 5.10)

### ISSUE THREAD

**Issue by bamboozila** [prisma/prisma#5304]
## Bug description

@prisma/cli installation hangs indefinitely inside of docker

## How to reproduce

REPRO: https://github.com/svelters/prisma-poc

## Expected behavior

It should install without container discrimination

## Prisma information

`@prisma/cli` version: "2.15.0"

## Environment & setup

- OS: Linux - Docker (alpine, debian etc...)
- Database: doesn't matter
- Node.js version: 15.6.0 but it seems to fail on any version we tried
- Prisma version: doesn't matter as it fails because of the cli

**Comment by Jolg42** [comment]
@bamboozila Could you try running `npm install --timing` maybe?

See https://docs.npmjs.com/generating-and-locating-npm-debug.log-files

It should output more logs and it will be easier to debug what's happening.

**Comment by bamboozila** [comment]
sure:

```
npm http fetch GET 200 https://registry.npmjs.org/graphql/-/graphql-15.5.0.tgz 4961ms
npm timing reifyNode:node_modules/graphql Completed in 6249ms
npm http fetch GET 200 https://registry.npmjs.org/es-abstract/-/es-abstract-1.18.0-next.2.tgz 6148ms
npm timing reifyNode:node_modules/es-abstract Completed in 6322ms
npm http fetch GET 200 https://registry.npmjs.org/@prisma/cli/-/cli-2.15.0.tgz 5033ms
npm timing reifyNode:node_modules/@prisma/cli Completed in 6337ms
npm http fetch GET 200 https://registry.npmjs.org/lodash/-/lodash-4.17.20.tgz 5773ms
npm timing reifyNode:node_modules/lodash Completed in 6407ms
npm http fetch GET 200 https://registry.npmjs.org/core-js/-/core-js-3.8.3.tgz 5582ms
npm timing reifyNode:node_modules/core-js Completed in 6586ms
npm http fetch GET 200 https://registry.npmjs.org/typescript/-/typescript-4.1.3.tgz 5570ms
npm timing reifyNode:node_modules/typescript Completed in 6840ms
npm timing reify:unpack Completed in 6854ms
npm timing reify:unretire Completed in 0ms
npm timing build:queue Completed in 9ms
npm timing build:run:preinstall:node_modules/@prisma/cli Completed in 59ms
npm timing build:run:preinstall Completed in 60ms
npm timing build:link:node_modules/@apollo/protobufjs Completed in 7ms
npm timing build:link:node_modules/graphql-tools/node_modules/uuid Completed in 7ms
npm timing build:link:node_modules/mime Completed in 8ms
npm timing build:link:node_modules/normalize-package-data/node_modules/semver Completed in 7ms
npm timing build:link:node_modules/dateformat Completed in 8ms
npm timing build:link:node_modules/mkdirp Completed in 9ms
npm timing build:link:node_modules/rimraf Completed in 9ms
npm timing build:link:node_modules/semver Completed in 9ms
npm timing build:link:node_modules/sha.js Completed in 9ms
npm timing build:link:node_modules/strip-indent Completed in 9ms
npm timing build:link:node_modules/tree-kill Completed in 9ms
npm timing build:link:node_modules/@prisma/cli Completed in 9ms
npm timing build:link:node_modules/ts-node Completed in 9ms
npm timing build:link:node_modules/ts-node-dev Completed in 9ms
npm timing build:link:node_modules/typescript Completed in 8ms
npm timing build:link:node_modules/uuid Completed in 8ms
npm timing build:link:node_modules/xss Completed in 8ms
npm timing build:link Completed in 11ms
npm timing build:run:install:node_modules/@prisma/cli Completed in 49ms
npm timing build:run:install Completed in 49ms
npm timing build:run:postinstall:node_modules/@apollo/protobufjs Completed in 62ms
npm timing build:run:postinstall:node_modules/core-js Completed in 76ms
npm timing build:run:postinstall:node_modules/nexus-plugin-prisma Completed in 74ms
npm timing build:run:postinstall:node_modules/@prisma/engines Completed in 8489ms

```

**Comment by Jolg42** [comment]
🤔 what's the `node -v` & `npm -v` version you have here?

**Comment by bamboozila** [comment]
`node -v` -> v15.6.0
`npm -v` -> 7.4.0

**Comment by mureni** [comment]
I had no issues with installation of @prisma/cli, however I did have a similar "indefinite hang" situation when trying to run `prisma generate` while using VSCode Remote Containers. I am not sure if this is related to the same docker problem, as there are no logs or debug items generated by this aspect of the Prisma CLI (at least, none that are documented).

**Comment by bamboozila** [comment]
Could someone try the repro?

It's minimal and I doubt it's my system causing this.

**Comment by Jolg42** [comment]
@bamboozila I'm not sure why I get this error when doing `make dev` 🤔

```
docker-compose build
Building api
Traceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 67, in main
  File "compose/cli/main.py", line 126, in perform_command
  File "compose/cli/main.py", line 302, in build
  File "compose/project.py", line 468, in build
  File "compose/project.py", line 450, in build_service
  File "compose/service.py", line 1125, in build
  File "site-packages/docker/api/build.py", line 261, in build
  File "site-packages/docker/api/build.py", line 308, in _set_auth_headers
  File "site-packages/docker/auth.py", line 311, in get_all_credentials
  File "site-packages/docker/auth.py", line 262, in _resolve_authconfig_credstore
  File "site-packages/docker/auth.py", line 287, in _get_store_instance
  File "site-packages/docker/credentials/store.py", line 25, in __init__
docker.credentials.errors.InitializationError: docker-credential-gcloud not installed or not available in PATH
[57943] Failed to execute script docker-compose
make: *** [dev] Error 255
```

**Comment by bamboozila** [comment]
> @bamboozila I'm not sure why I get this error when doing `make dev` 
> 
> Seems Google Cloud SDK is required but I don't know why.
> 
> ```
> docker-compose build
> Building api
> Traceback (most recent call last):
>   File "docker-compose", line 3, in <module>
>   File "compose/cli/main.py", line 67, in main
>   File "compose/cli/main.py", line 126, in perform_command
>   File "compose/cli/main.py", line 302, in build
>   File "compose/project.py", line 468, in build
>   File "compose/project.py", line 450, in build_service
>   File "compose/service.py", line 1125, in build
>   File "site-packages/docker/api/build.py", line 261, in build
>   File "site-packages/docker/api/build.py", line 308, in _set_auth_headers
>   File "site-packages/docker/auth.py", line 311, in get_all_credentials
>   File "site-packages/docker/auth.py", line 262, in _resolve_authconfig_credstore
>   File "site-packages/docker/auth.py", line 287, in _get_store_instance
>   File "site-packages/docker/credentials/store.py", line 25, in __init__
> docker.credentials.errors.InitializationError: docker-credential-gcloud not installed or not available in PATH
> [57943] Failed to execute script docker-compose
> make: *** [dev] Error 255
> ```

https://github.com/docker/compose/issues/5930

It seems you have to add `export CLOUDSDK_PYTHON=python2` to your `~/.bashrc` / `~/.zshrc`

**Comment by Jolg42** [comment]
After installing gcloud sdk it seems that the build works fine for me.

```
>make dev
docker-compose build
db uses an image, skipping
Building api
Step 1/7 : FROM node:15-alpine as base
 ---> 706d12284dd5
Step 2/7 : WORKDIR /app
 ---> Using cache
 ---> 9b3b82d9c6d0

Step 3/7 : FROM base as dev
 ---> 9b3b82d9c6d0
Step 4/7 : ENV NODE_ENV development
 ---> Using cache
 ---> 082088e7d861
Step 5/7 : USER node
 ---> Using cache
 ---> 117a94d96926
Step 6/7 : COPY . /app
 ---> bd85f4efe9fe
Step 7/7 : CMD ["yarn", "dev"]
 ---> Running in 36d7006ae5ed
Removing intermediate container 36d7006ae5ed
 ---> af4d72abd0f9

Successfully built af4d72abd0f9
Successfully tagged prisma-poc_api:latest
docker-compose run -T api yarn install
Creating network "prisma-poc_default" with the default driver
Creating prisma-poc_api_run ... done
yarn install v1.22.5
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.30s.
docker-compose up -d
Creating prismapoc-api ... done
Creating prismapoc-db  ... done
```

**Comment by Jolg42** [comment]
Maybe worth trying `docker-compose build --no-cache` ?

**Comment by dickfickling** [comment]
@bamboozila I ran into the same issue. obviously not a great solution, but i worked around this by waiting to copy my prisma schema into the docker image until after running `npm install`:

```
FROM node:14-alpine AS builder

ARG NODE_ENV
ENV NODE_ENV "$NODE_ENV"
ENV PATH $PATH:/opt/app/node_modules/.bin

RUN mkdir -p /opt/app/build
WORKDIR /opt/app

COPY package.json /opt/app
COPY package-lock.json /opt/app
RUN npm install

COPY prisma /opt/app/prisma
COPY tsconfig.json /opt/app/tsconfig.json
COPY src /opt/app/src
COPY config /opt/app/config
RUN node_modules/.bin/prisma generate
RUN node_modules/.bin/tsc
```

It'll show the following warning in the `npm install` output, but it's fine because we run `generate` manually later
```warning The postinstall script automatically ran `prisma generate` and did not find your `prisma/schema.prisma`.```

**Comment by bamboozila** [comment]
Ok, so the issue comes from the kernel version used.

* No issue on a debian stretch VM using version `4.19.0-13`.
* The issue is present on version `5.10.0-2`. As is it linked to the kernel, it does affect any linux distribution.

I can confirm the culprit is https://github.com/prisma/prisma/blob/master/src/packages/client/scripts/postinstall.js

Thank you @dickfickling 

```js
  try {
    if (localPath) {
      await run('node', [
        localPath,
        'generate',
        '--postinstall',
        doubleQuote(getPostInstallTrigger()),
      ])
      return
    }
```

I spent a lot of time trying to pinpoint the issue.

Maybe someone working on `@prisma/cli` could take a look?

**Comment by Jolg42** [comment]
Interesting! Thanks for the new details.

So to confirm there is no log, and it hangs at `prisma generate` executed during postinstall? (Tested with @prisma/cli 2.15.0?)

A workaround / way to confirm the problem would be to set the env var `SKIP_GENERATE="true"` which will skip that generate on postinstall. Did you try that maybe?

**Comment by Jolg42** [comment]
Could you try again with `@prisma/cli` 2.16.0? It's likely fixed now.

**Comment by bamboozila** [comment]
just tried it, it's still broken.

I've updated the repro though.

**Comment by alechp** [comment]
@matthewmueller Saw you removed the 2.17 milestone. Is this being targeted in 2.18?

**Comment by bamboozila** [comment]
the repro has been updated and still failing on 2.17.0.

Both prisma (cli) and @prisma/client have been updated.

**Comment by Jolg42** [comment]
Just did a `git pull` tried and it seems the install works at least 

```
j42@Pluto ~/D/r/prisma-poc> make clean                                                              docker-compose down -v
Removing network prisma-poc_default
WARNING: Network prisma-poc_default not found.
rm -rf ./apps/api/node_modules
j42@Pluto ~/D/r/prisma-poc> make dev
echo "build"
build
docker-compose build
db uses an image, skipping
Building api


Command killed by keyboard interrupt

Step 1/7 : FROM node:15-alpine as base
 ---> 706d12284dd5
Step 2/7 : WORKDIR /app
 ---> Using cache
 ---> 9b3b82d9c6d0

Step 3/7 : FROM base as dev
 ---> 9b3b82d9c6d0
Step 4/7 : ENV NODE_ENV development
 ---> Using cache
 ---> 082088e7d861
Step 5/7 : USER node
 ---> Using cache
 ---> 117a94d96926
Step 6/7 : COPY . /app
 ---> 9132e9a43669
Step 7/7 : CMD ["yarn", "dev"]
 ---> Running in e4627f614639
Removing intermediate container e4627f614639
 ---> 542527fe9d37

Successfully built 542527fe9d37
Successfully tagged prisma-poc_api:latest
echo "run"
run
docker-compose run -T api yarn install
Creating network "prisma-poc_default" with the default driver
Creating prisma-poc_api_run ... done
yarn install v1.22.5
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@2.3.1: The platform "linux" is incompatible with this module.
info "fsevents@2.3.1" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > apollo-server@2.19.2" has unmet peer dependency "graphql@^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0".
warning "apollo-server > apollo-server-core@2.19.2" has unmet peer dependency "graphql@^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0".
warning "apollo-server > apollo-server-express@2.19.2" has unmet peer dependency "graphql@^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0".
warning "apollo-server > graphql-subscriptions@1.1.0" has unmet peer dependency "graphql@^0.10.5 || ^0.11.3 || ^0.12.0 || ^0.13.0 || ^14.0.0".
warning "apollo-server > graphql-tools@4.0.8" has unmet peer dependency "graphql@^0.13.0 || ^14.0.0 || ^15.0.0".
warning "apollo-server > apollo-server-core > apollo-cache-control@0.11.6" has unmet peer dependency "graphql@^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0".
warning "apollo-server > apollo-server-core > apollo-graphql@0.6.0" has unmet peer dependency "graphql@^14.2.1 || ^15.0.0".
warning "apollo-server > apollo-server-core > apollo-server-errors@2.4.2" has unmet peer dependency "graphql@^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0".
warning "apollo-server > apollo-server-core > apollo-server-plugin-base@0.10.4" has unmet peer dependency "graphql@^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0".
warning "apollo-server > apollo-server-core > apollo-server-types@0.6.3" has unmet peer dependency "graphql@^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0".
warning "apollo-server > apollo-server-core > apollo-tracing@0.12.2" has unmet peer dependency "graphql@^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0".
warning "apollo-server > apollo-server-core > graphql-extensions@0.12.8" has unmet peer dependency "graphql@^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0".
warning "apollo-server > apollo-server-core > graphql-tag@2.11.0" has unmet peer dependency "graphql@^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0".
warning "apollo-server > apollo-server-core > graphql-upload@8.1.0" has unmet peer dependency "graphql@0.13.1 - 14".
warning "apollo-server > apollo-server-core > subscriptions-transport-ws@0.9.18" has unmet peer dependency "graphql@>=0.10.0".
warning "apollo-server > graphql-tools > apollo-link@1.2.14" has unmet peer dependency "graphql@^0.11.3 || ^0.12.3 || ^0.13.0 || ^14.0.0 || ^15.0.0".
warning "apollo-server > graphql-tools > apollo-utilities@1.3.4" has unmet peer dependency "graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0".
warning " > nexus@1.0.0" has unmet peer dependency "graphql@^14.5.8 || 15.x".
warning " > nexus-plugin-prisma@0.28.0" has incorrect peer dependency "@prisma/client@2.14.x".
warning " > nexus-plugin-prisma@0.28.0" has unmet peer dependency "graphql@^15.3.0".
[4/4] Building fresh packages...
Done in 69.98s.
echo "up"
up
docker-compose up -d
Creating prismapoc-db ... 
Creating prismapoc-db  ... error
WARNING: Host is already in use by another container

Creating prismapoc-api ... done

ERROR: for db  Cannot start service db: driver failed programming external connectivity on endpoint prismapoc-db (38c4f78d4294d638f65c7f83cd364386a45c3a92d81625b678a5626bafb7b011): Bind for 0.0.0.0:5432 failed: port is already allocated
ERROR: Encountered errors while bringing up the project.
```
It also works if I do `docker-compose down -v --rmi all --remove-orphans` before.

Notes:
- it took 69.98s for the build steps for me where dependencies are downloaded / installed.
- the db errors because I already have a postgres instance locally but that seems irrelevant to this issue
- nexus-plugin-prisma doesn't support 2.17.0 at the moment, only 2.16.0, see https://github.com/graphql-nexus/nexus-plugin-prisma/releases I tried having those versions in sync here, and it works as well https://github.com/Jolg42/prisma-poc/commit/704741aea1afa5d427f955ec1bc600ecb287ecd3

Maybe @pantharshit00 can try a reproduction on Linux, I'm on macOS Catalina 10.15.7, docker-compose version 1.27.4, build 40524192.

I did another run, it's noticeable that something it's stuck after 
`warning " > nexus-plugin-prisma@0.30.0" has unmet peer dependency "graphql@^15.3.0".`
for a while but stills seems to resolves after some waiting with
```
[4/4] Building fresh packages...
Done in 68.82s.
```

**Comment by Jolg42** [comment]
@bamboozila could you provide the raw logs you see maybe?
Do you have a specific network configuration? (proxy, Chinese Great Firewall...)

**Comment by bamboozila** [comment]
in order to reproduce the issue, you have to use a 5+ linux kernel.
It doesn't surprise me it works fine on mac as the hypervisor does a different job.

My network is quiet normal, no special stuff going on.
The issue is just the same at work too.

Regarding the logs, there isn't much more going on.

Could someone at prisma try with a debian "unstable" VM?

This is the one I used to pinpoint the kernel issue: 
https://cdimage.debian.org/cdimage/weekly-builds/amd64/iso-cd/debian-testing-amd64-netinst.iso

**Comment by alechp** [comment]
I'm noticing the same (or related) issue. Hopefully helpful context:

- I currently have a successfully deployed prisma@2.6.1 service running in the wild. Works locally (both node & docker), passes CI/CD build step and is successfully running in production (docker)

- I have a new prisma service that runs locally (both node & docker), but fails when compiling in GitHub Actions

These services are very similar.
- Both work perfectly when run locally (in either raw nodeJS or inside docker)
- Both have identical Dockerfiles
- Both are using the same CI/CD pipeline (ie. Identical GitHub Actions script, both using Docker Hub, both being deployed to the same Gateway on k8s, same secret management, etc.)

The only differences: 
- new service fails to deploy using the same GitHub Actions script
- new service is running prisma 2.17.0 (2.16.1 fails as well) instead of the older 2.6.1
- new service is using Nexus with prisma
- new service is using the express Apollo Server along with the vanilla express package

**Comment by bamboozila** [comment]
@Jolg42 Would it be possible to include this issue in the [2.18.0 Milestone](https://github.com/prisma/prisma/milestone/65)?

**Comment by Jolg42** [comment]
I added the candidate label so we don't forget about it, I think @pantharshit00 has this on his list to try but I think he's quite busy right now.
🤞

**Comment by alechp** [comment]
@bamboozila Seems like this is going to be a hard blocker for a while. 

Couple questions:
1. What's the latest working version of prisma you've found? I've established 2.6.1 works fine, but I imagine there's a more recent working version 

2. Are you using Nexus?

**Comment by bamboozila** [comment]
@alechp 

1. Nice find, the latest working version is the 10.0.0
The 10.0.1 is the culprit! @Jolg42 https://github.com/prisma/prisma/commit/ad8506c1e01d52f574484461055aeb4599c2b929
It seems indeed related to the @prisma/engine

2. Yup but I am only at the bootstrap stage so far on my project (looking into authentication workflow with nexus)

**Comment by Jolg42** [comment]
Just checked and it appears we changed build images used to produce the binaries one day before that specific commit https://github.com/prisma/engine-images/commits/master
<img width="977" alt="Screen Shot 2021-02-24 at 17 44 01" src="https://user-images.githubusercontent.com/1328733/109034486-012a0580-76c8-11eb-8653-52346be30653.png">

Likely related I guess.

**Comment by pantharshit00** [comment]
Hey sorry for the delay here. I was busy with my exams. I will look into this today :)

**Comment by pantharshit00** [comment]
I can reproduce this in a debian vm which I manually upgraded to linux kernal 5.10 on digitalocean.

![image](https://user-images.githubusercontent.com/22195362/109311255-97e2f780-786b-11eb-816f-8c5622ed8bc5.png)
```
Linux debian-s-1vcpu-1gb-intel-blr1-01 5.10.17-051017-generic #202102170631 SMP Wed Feb 17 11:37:41 UTC 2021 x86_64 GNU/Linux
```

This will need more investigation. @pimeys should look into this as @Jolg42 pointed out this might be build related.

**Comment by bamboozila** [comment]
I have a good workaround:

https://github.com/svelters/prisma-poc/commit/a21cf21b1172670c0222b5130c69428c7438e04d

It seems like the problem is tty related.

**Comment by Jolg42** [comment]
@bamboozila so removing `-T` from `docker-compose run -T api yarn install` fixes your issue 🤔 
> -T Disable pseudo-tty allocation. By default `docker-compose run` allocates a TTY.

**Comment by bamboozila** [comment]
Yes it does but it shouldn't be needed.

**Comment by alechp** [comment]
Related?

https://github.com/prisma/prisma-client-go/issues/357

**Comment by sayerrrr** [comment]
I'm having a similar issue running npm i in a codebuild project.. any official workaround?

**Comment by Jolg42** [comment]
@sayerrrr Interesting! Could you open an issue so we could try to reproduce?

**Comment by neilime** [comment]
Same issue while running `prisma generate` in a `node:lts-alpine` container.
It works in a `node:lts` container.

It appears to be an issue with alpine images

**Comment by pantharshit00** [comment]
@neilime Can you share your image code? That might help us in fixing this.

**Comment by piec** [comment]
Hi all,

I'm reproducing the issue on a debian 10.9 machine. I'm using the latest version of the reproduction https://github.com/svelters/prisma-poc/commit/a21cf21b1172670c0222b5130c69428c7438e04d . Removing `-T` does not work for me.
```
Linux 5.10.0-0.bpo.4-amd64 #1 SMP Debian 5.10.19-1~bpo10+1 (2021-03-13) x86_64 GNU/Linux
```

`docker-compose run api yarn install` is stuck and here is what's happening in the container:
```
% docker exec -it --user 0 cf9d941a9677 sh -c 'apk add procps; ps auxf | cat'
...
node         1 12.9  0.3 366632 124192 pts/0   Ssl+ 21:42   0:19 node /opt/yarn-v1.22.5/bin/yarn.js install --verbose
node        65  0.0  0.1 280304 33532 ?        Ssl  21:43   0:00 /usr/local/bin/node scripts/postinstall.js
node        76  100  0.2 345176 69076 ?        Sl   21:43   2:20  \_ /usr/local/bin/node /app/node_modules/prisma/build/index.js generate --postinstall "yarn install --verbose"
```
Notice 100% cpu, strace shows 0 syscall activity.

---

The exact same thing works on another Arch Linux machine (updated ~a week ago).
```
Linux 5.11.13-arch1-1 #1 SMP PREEMPT Sat, 10 Apr 2021 20:47:14 +0000 x86_64 GNU/Linux
```

very weird

**Comment by piec** [comment]
On this debian 10.9 machine I noticed that using <= node:13-alpine works:
* node:12-alpine: ok
* node:13-alpine: ok
* node:14-alpine: bad
* node:15-alpine: bad

**Comment by carlnc** [comment]
uname -a

```
Linux 2ef0c037-7695-4298-81e5-1616836bacbf-ppkrl 5.10.25-flatcar #1 SMP Wed Mar 24 17:11:24 -00 2021 x86_64 Linux
```

```
    export DEBUG="prisma,prisma:*"
```

first time `pnpm install --filter="{.}..."` :
```
.../node_modules/@prisma/engines postinstall$ node download/index.js
../..                                    | Progress: resolved 1017, reused 0, downloaded 1017, added 1017, done
.../nodemon@2.0.7/node_modules/nodemon postinstall$ node bin/postinstall || exit 0
.../nodemon@2.0.7/node_modules/nodemon postinstall: Done
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:24.501Z prisma:download file /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/query-engine-linux-musl does not exist and must be downloaded
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:24.503Z prisma:download file /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/migration-engine-linux-musl does not exist and must be downloaded
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:24.503Z prisma:download file /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/prisma-fmt-linux-musl does not exist and must be downloaded
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:24.503Z prisma:download file /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/introspection-engine-linux-musl does not exist and must be downloaded
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:24.506Z prisma:download Downloading https://binaries.prisma.sh/all_commits/cd459a60adc2f40ba38ed4cb1d6a5885e452c515/linux-musl/query-engine.gz to /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/query-engine-linux-musl
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:24.536Z prisma:download Downloading https://binaries.prisma.sh/all_commits/cd459a60adc2f40ba38ed4cb1d6a5885e452c515/linux-musl/migration-engine.gz to /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/migration-engine-linux-musl
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:24.538Z prisma:download Downloading https://binaries.prisma.sh/all_commits/cd459a60adc2f40ba38ed4cb1d6a5885e452c515/linux-musl/introspection-engine.gz to /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/introspection-engine-linux-musl
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:24.539Z prisma:download Downloading https://binaries.prisma.sh/all_commits/cd459a60adc2f40ba38ed4cb1d6a5885e452c515/linux-musl/prisma-fmt.gz to /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/prisma-fmt-linux-musl
.../node_modules/@prisma/engines postinstall: Done
.../node_modules/prisma preinstall$ node scripts/preinstall-entry.js
.../node_modules/prisma preinstall: Done
.../node_modules/prisma install$ node scripts/install-entry.js
.../node_modules/prisma install: Done
.../node_modules/@prisma/client postinstall$ node scripts/postinstall.js
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:26.832Z prisma:loadEnv project root found at /opt/atlassian/pipelines/agent/build/node/package.json
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:26.845Z prisma:tryLoadEnv Environment variables not found at null
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:26.845Z prisma:tryLoadEnv Environment variables not found at undefined
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:26.846Z prisma:tryLoadEnv No Environment variables loaded
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:26.847Z prisma:engines using NAPI: false
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:26.848Z prisma:engines binaries to download query-engine, migration-engine, introspection-engine, prisma-fmt
.../node_modules/@prisma/client postinstall: prisma:warn The postinstall script automatically ran `prisma generate` and did not find your `prisma/schema.prisma`.
.../node_modules/@prisma/client postinstall: If you have a Prisma schema file in a custom path, you will need to run
.../node_modules/@prisma/client postinstall: `prisma generate --schema=./path/to/your/schema.prisma` to generate Prisma Client.
.../node_modules/@prisma/client postinstall: If you do not have a Prisma schema file yet, you can ignore this message.
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:27.697Z prisma:cli Error: Could not find a schema.prisma file that is required for this command.
.../node_modules/@prisma/client postinstall: You can either provide it with --schema, set it as `prisma.schema` in your package.json or put it into the default location ./prisma/schema.prisma https://pris.ly/d/prisma-schema-location
.../node_modules/@prisma/client postinstall:     at Object.getSchema3 (/opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/prisma@2.23.0-dev.7/node_modules/prisma/build/index.js:10136:13)
.../node_modules/@prisma/client postinstall:     at async main (/opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/prisma@2.23.0-dev.7/node_modules/prisma/build/index.js:109780:22)
.../node_modules/@prisma/client postinstall: Done
```

after: 
```bash
rm -rf node_modules ../../node_modules &&
     rm -rf /root/.cache/prisma &&
     pnpm install --no-optional --prod --filter="{.}..."
```

```
.../node_modules/@prisma/engines postinstall$ node download/index.js
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:54.058Z prisma:download file /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/migration-engine-linux-musl does not exist and must be downloaded
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:54.060Z prisma:download file /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/introspection-engine-linux-musl does not exist and must be downloaded
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:54.060Z prisma:download file /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/query-engine-linux-musl does not exist and must be downloaded
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:54.060Z prisma:download file /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/prisma-fmt-linux-musl does not exist and must be downloaded
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:54.063Z prisma:download Downloading https://binaries.prisma.sh/all_commits/cd459a60adc2f40ba38ed4cb1d6a5885e452c515/linux-musl/query-engine.gz to /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/query-engine-linux-musl
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:54.101Z prisma:download Downloading https://binaries.prisma.sh/all_commits/cd459a60adc2f40ba38ed4cb1d6a5885e452c515/linux-musl/introspection-engine.gz to /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/introspection-engine-linux-musl
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:54.103Z prisma:download Downloading https://binaries.prisma.sh/all_commits/cd459a60adc2f40ba38ed4cb1d6a5885e452c515/linux-musl/migration-engine.gz to /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/migration-engine-linux-musl
.../node_modules/@prisma/engines postinstall: 2021-05-07T02:07:54.105Z prisma:download Downloading https://binaries.prisma.sh/all_commits/cd459a60adc2f40ba38ed4cb1d6a5885e452c515/linux-musl/prisma-fmt.gz to /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/@prisma+engines@2.23.0-6.cd459a60adc2f40ba38ed4cb1d6a5885e452c515/node_modules/@prisma/engines/prisma-fmt-linux-musl
.../node_modules/@prisma/engines postinstall: Done
.../node_modules/prisma preinstall$ node scripts/preinstall-entry.js
.../node_modules/prisma preinstall: Done
.../node_modules/prisma install$ node scripts/install-entry.js
.../node_modules/prisma install: Done
.../node_modules/@prisma/client postinstall$ node scripts/postinstall.js
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:58.155Z prisma:loadEnv project root found at /opt/atlassian/pipelines/agent/build/node/package.json
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:58.171Z prisma:tryLoadEnv Environment variables not found at null
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:58.171Z prisma:tryLoadEnv Environment variables not found at undefined
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:58.171Z prisma:tryLoadEnv No Environment variables loaded
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:58.173Z prisma:engines using NAPI: false
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:58.173Z prisma:engines binaries to download query-engine, migration-engine, introspection-engine, prisma-fmt
.../node_modules/@prisma/client postinstall: prisma:warn The postinstall script automatically ran `prisma generate` and did not find your `prisma/schema.prisma`.
.../node_modules/@prisma/client postinstall: If you have a Prisma schema file in a custom path, you will need to run
.../node_modules/@prisma/client postinstall: `prisma generate --schema=./path/to/your/schema.prisma` to generate Prisma Client.
.../node_modules/@prisma/client postinstall: If you do not have a Prisma schema file yet, you can ignore this message.
.../node_modules/@prisma/client postinstall: 2021-05-07T02:07:59.323Z prisma:cli Error: Could not find a schema.prisma file that is required for this command.
.../node_modules/@prisma/client postinstall: You can either provide it with --schema, set it as `prisma.schema` in your package.json or put it into the default location ./prisma/schema.prisma https://pris.ly/d/prisma-schema-location
.../node_modules/@prisma/client postinstall:     at Object.getSchema3 (/opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/prisma@2.23.0-dev.7/node_modules/prisma/build/index.js:10136:13)
.../node_modules/@prisma/client postinstall:     at async main (/opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/prisma@2.23.0-dev.7/node_modules/prisma/build/index.js:109780:22)
.../node_modules/@prisma/client postinstall: Done
```

and then the kicker: `pnpx prisma generate`

```
2021-05-07T02:08:00.429Z prisma:loadEnv project root found at /opt/atlassian/pipelines/agent/build/node/Services/service-xxxxxxxxxx/package.json
2021-05-07T02:08:00.442Z prisma:tryLoadEnv Environment variables not found at null
2021-05-07T02:08:00.443Z prisma:tryLoadEnv Environment variables not found at undefined
2021-05-07T02:08:00.443Z prisma:tryLoadEnv No Environment variables loaded
2021-05-07T02:08:00.445Z prisma:engines using NAPI: false
2021-05-07T02:08:00.445Z prisma:engines binaries to download query-engine, migration-engine, introspection-engine, prisma-fmt
Prisma schema loaded from prisma/schema.prisma
2021-05-07T02:08:01.684Z prisma:download copying /root/.cache/prisma/master/cd459a60adc2f40ba38ed4cb1d6a5885e452c515/linux-musl/query-engine to /opt/atlassian/pipelines/agent/build/node/node_modules/.pnpm/prisma@2.23.0-dev.7/node_modules/prisma/query-engine-linux-musl
```

Entrypoint for [node download/index.js](https://github.com/prisma/engines-wrapper/blob/master/packages/engines/src/download.ts#L13)

[@prisma/fetch-engine "prisma:download copying ... to ..." ](https://github.com/prisma/prisma/blob/master/src/packages/fetch-engine/src/download.ts#L292)

Perhaps [copy](https://github.com/prisma/prisma/blob/master/src/packages/fetch-engine/src/copy.ts) or [getHash](https://github.com/prisma/prisma/blob/14ffd0093c1b8c7db1f45e9f04c5fab7c1c120d2/src/packages/fetch-engine/src/getHash.ts#L4) never return?


....

I have managed to get past it by calling `rm -rf /root/.cache/prisma/` before each `npm install` / `pnpx prisma generate`

**Comment by ywada526** [comment]
I had the same issue with `docker-compose run --rm api yarn install` on a Mac machine.

```
$ uname -a
Darwin MacBook-Pro.local 20.4.0 Darwin Kernel Version 20.4.0: Thu Apr 22 21:46:41 PDT 2021; root:xnu-7195.101.2~1/RELEASE_ARM64_T8101 arm64
```

https://github.com/prisma/prisma/issues/5304#issuecomment-824387989 is also true in my environment,
It's stuck with node:15-buster, but not with node:10-buster.

And, once I complete `yarn install` with node:10-buster and save `node_modules` to volume, it won't be stuck even at 15.

**Comment by piec** [comment]
@ywada526 you should run uname in your container. As far as I know docker containers run in a virtualized  linux on macos

**Comment by ywada526** [comment]
@piec

In my docker container:
```
$ uname -a
Linux b2e62a121b86 5.10.25-linuxkit #1 SMP PREEMPT Tue Mar 23 09:24:45 UTC 2021 aarch64 GNU/Linux
```

**Comment by janpio** [comment]
For everyone looking for a workaround that can not upgrade their Kernel version:

Note the last sentence in carlnc's comment above: 

> I have managed to get past it by calling `rm -rf /root/.cache/prisma/` before each `npm install` / `pnpx prisma generate`

https://github.com/prisma/prisma/issues/5304#issuecomment-834019800

**Comment by moderndegree** [comment]
Same issue here.

In my docker container:
```
$ uname -a
Linux 6ac4dcf2264a 5.10.25-linuxkit #1 SMP Tue Mar 23 09:27:39 UTC 2021 x86_64 GNU/Linux
```
It might be worth mentioning that I am using the VS Code Remote Containers extension for development. I am using the Node with Postgres container.

**Comment by JacobMuchow** [comment]
Same issue here :/ It is currently making it difficult for my team to adopt Prisma because we all run local dev environments using docker compose. I like Prisma and really want to keep working with it but it's a back breaker at the moment.

docker-compose.yml

```
version: "3.2"
services:
    environment:
      - DATABASE_URL=postgresql://blahblah
    ports:
      - "8001:8001"
    volumes:
      - ./backend:/app
      - /app/node_modules
```

Dockerfile

```
FROM node:14.17.0

RUN yarn set version berry

WORKDIR /app

CMD ./scripts/start.sh
```

scripts/start.sh

```
yarn

npx prisma migrate deploy
npx prisma generate

yarn start-dev
```

Tried adding the `rm -rf /root/.cache/prisma/` line before `yarn` but no dice. 

Yarn hangs on the Link step at this line:

```
backend_1            | ➤ YN0007: │ prisma@npm:2.21.2 must be built because it never did before or the last one failed
backend_1            | ➤ YN0007: │ @prisma/client@npm:2.22.1 [b7322] must be built because it never did before or the last one failed
```

One build log I got from this was:

```
# This file contains the result of Yarn building a package (@prisma/client@virtual:b73228cf63bebf8a7b2e8c303885b6a2b3599c5aaddeba740806f8b2886d720f8e239abc5a32bf74e53c0aa16788884f3fbd064b45ca1e62b72c85c745283b9b#npm:2.22.1)
# Script name: postinstall

Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
```

Will update if I learn more on my own

**Comment by carlnc** [comment]
@JacobMuchow 

Are you running as root?

Did you use `export DEBUG="prisma,prisma:*"` to make sure that `/root/.cache/prisma/` is the location that `post-install` is saving the file to?

**Comment by JacobMuchow** [comment]
@carlnc thanks for the tip! Looks like it. This is where it is hanging currently:

```
Environment variables loaded from .env
prisma:engines using NAPI: false
prisma:engines binaries to download query-engine, migration-engine, introspection-engine, prisma-fmt
Prisma schema loaded from prisma/schema.prisma
prisma:download copying /root/.cache/prisma/master/e421996c87d5f3c8f7eeadd502d4ad402c89464d/debian-openssl-1.1.x/query-engine to /app/node_modules/prisma/query-engine-debian-openssl-1.1.x
```

**Comment by carlnc** [comment]
@JacobMuchow 

Did you try adding `rm -rf /root/.cache/prisma/` before

both: 

* `yarn`, and 
*  `npx prisma generate`

You should be finding that:

* The 1st time, yarn /  prisma / post-install ... downloads
* The 2nd time, prisma generate ... does a COPY.

It's the COPY, that is falling over.

**Comment by JacobMuchow** [comment]
I have, yeah. It looks to be the same result from the prisma post-install script hanging on the download/copy line:

```
prisma:download copying /root/.cache/prisma/master/e421996c87d5f3c8f7eeadd502d4ad402c89464d/debian-openssl-1.1.x/query-engine to /app/node_modules/prisma/query-engine-debian-openssl-1.1.x
```

**Comment by JacobMuchow** [comment]
Here's some more info from a full verbose output from npm. 

```
npm info lifecycle @prisma/engines@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d~postinstall: @prisma/engines@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d

> @prisma/engines@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d postinstall /app/node_modules/@prisma/engines
> node download/index.js

  prisma:download file /app/node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x does not exist and must be downloaded +0ms
  prisma:download file /app/node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x does not exist and must be downloaded +2ms
  prisma:download file /app/node_modules/@prisma/engines/query-engine-debian-openssl-1.1.x does not exist and must be downloaded +0ms
  prisma:download file /app/node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x does not exist and must be downloaded +0ms
  prisma:download Downloading https://binaries.prisma.sh/all_commits/e421996c87d5f3c8f7eeadd502d4ad402c89464d/debian-openssl-1.1.x/query-engine.gz to /app/node_modules/@prisma/engines/query-engine-debian-openssl-1.1.x +2ms
> Downloading Prisma engines for debian-openssl-1.1.x [                    ] 0%  prisma:download Downloading https://binaries.prisma.sh/all_commits/e421996c87d5f3c8f7eeadd502d4ad402c89464d/debian-openssl-1.1.x/migration-engine.gz to /app/node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x +16ms
  prisma:download Downloading https://binaries.prisma.sh/all_commits/e421996c87d5f3c8f7eeadd502d4ad402c89464d/debian-openssl-1.1.x/introspection-engine.gz to /app/node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x +2ms
  prisma:download Downloading https://binaries.prisma.sh/all_commits/e421996c87d5f3c8f7eeadd502d4ad402c89464d/debian-openssl-1.1.x/prisma-fmt.gz to /app/node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x +2ms
npm verb lifecycle @prisma/engines@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d~postinstall: unsafe-perm in lifecycle false
npm verb lifecycle @prisma/engines@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d~postinstall: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/app/node_modules/@prisma/engines/node_modules/.bin:/app/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
npm verb lifecycle @prisma/engines@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d~postinstall: CWD: /app/node_modules/@prisma/engines
npm info lifecycle @prisma/engines-version@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d~postinstall: @prisma/engines-version@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d


...
npm info lifecycle @prisma/client@2.21.1~postinstall: @prisma/client@2.21.1

> @prisma/client@2.21.1 postinstall /app/node_modules/@prisma/client
> node scripts/postinstall.js

  prisma:loadEnv project root found at /app/package.json +0ms
  prisma:tryLoadEnv Environment variables loaded from /app/.env +0ms
[dotenv][DEBUG] "DATABASE_URL" is already defined in `process.env` and will not be overwritten
Environment variables loaded from .env
  prisma:engines using NAPI: false +0ms
  prisma:engines binaries to download query-engine, migration-engine, introspection-engine, prisma-fmt +0ms
Prisma schema loaded from prisma/schema.prisma
  prisma:download copying /root/.cache/prisma/master/e421996c87d5f3c8f7eeadd502d4ad402c89464d/debian-openssl-1.1.x/query-engine to /app/node_modules/prisma/query-engine-debian-openssl-1.1.x +0ms
```

**Comment by janpio** [comment]
The script is really doing, or trying to do what is says here:

> ```
>   prisma:download copying /root/.cache/prisma/master/e421996c87d5f3c8f7eeadd502d4ad402c89464d/debian-openssl-1.1.x/query-engine to /app/node_modules/prisma/query-engine-debian-openssl-1.1.x +0ms
> ```

That this does not work makes absolutely no sense to us and we have no idea how we could possibly fix this.

**Comment by JacobMuchow** [comment]
I agree. At this line, it's copying this file from the container to the unnamed volume for my node_modules, so I am suspecting there is either:

1) something wrong with this concept that I am not understanding or
2) that I have encountered some bug with docker desktop for mac

https://github.com/docker/for-mac/issues/4957#issuecomment-843248099

I was seeing these issues on Friday after updating (though not this week), but digging into the problem it seems to be intermittent for various people with no clear version to roll back to. But I'll dig in more and post back if I find out. Want to help out if I can!

**Comment by JacobMuchow** [comment]
Downgrading Docker desktop for Mac to 2.3.0.5 worked for me.

I have tested and in my case do not need to do the `rm -rf /root/.cache/prisma/` at all now.

Seems like they have made some file system changes that are causing issues :/ thanks for the help guys

**Comment by JcdeA** [comment]
Having the same issue inside an Ubuntu container (tried debian, alpine, too) in an Ubuntu VM with Linux 5.8 (for Gitlab CI) ..
```
jcde@fridge:~$ uname -r
5.8.0-53-generic
```
Example:
```
$ yarn --frozen-lockfile --production
yarn install v1.22.10
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@2.3.2: The platform "linux" is incompatible with this module.
info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning "@geist-ui/react > styled-jsx@3.3.0" has incorrect peer dependency "react@15.x.x || 16.x.x".
warning "blitz > @testing-library/react-hooks@4.0.1" has incorrect peer dependency "react@>=16.9.0".
warning "blitz > react-test-renderer@17.0.1" has incorrect peer dependency "react@17.0.1".
warning "blitz > @blitzjs/babel-preset > babel-plugin-superjson-next@0.2.2" has unmet peer dependency "next@>=9.0.0".
warning "blitz > @blitzjs/babel-preset > babel-plugin-superjson-next@0.2.2" has unmet peer dependency "superjson@1.x".
warning "blitz > @blitzjs/core > next@10.1.3-0.34.3" has incorrect peer dependency "react@^16.6.0 || ^17".
warning "blitz > @blitzjs/core > next@10.1.3-0.34.3" has incorrect peer dependency "react-dom@^16.6.0 || ^17".
warning "blitz > @blitzjs/core > null-loader@4.0.1" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
warning "blitz > @blitzjs/core > react-query@3.13.9" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "blitz > @blitzjs/generator > jscodeshift@0.11.0" has unmet peer dependency "@babel/preset-env@^7.1.6".
warning "blitz > @blitzjs/installer > ink@3.0.8" has incorrect peer dependency "react@>=16.8.0".
warning "blitz > @blitzjs/installer > ink-spinner@4.0.1" has incorrect peer dependency "react@^16.8.2".
warning "blitz > react-test-renderer > react-shallow-renderer@16.14.1" has incorrect peer dependency "react@^16.0.0 || ^17.0.0".
warning "blitz > @blitzjs/core > next > @next/react-dev-overlay@10.1.3" has incorrect peer dependency "react@^16.9.0 || ^17".
warning "blitz > @blitzjs/core > next > @next/react-dev-overlay@10.1.3" has incorrect peer dependency "react-dom@^16.9.0 || ^17".
warning "blitz > @blitzjs/core > next > styled-jsx@3.3.2" has incorrect peer dependency "react@15.x.x || 16.x.x || 17.x.x".
warning "blitz > @blitzjs/core > next > use-subscription@1.5.1" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "blitz > @blitzjs/installer > ink > react-reconciler@0.24.0" has incorrect peer dependency "react@^16.0.0".
warning " > react-error-boundary@3.1.3" has incorrect peer dependency "react@>=16.13.1".
warning " > react-feather@2.0.9" has incorrect peer dependency "react@^16.8.6 || ^17".
warning " > react-final-form@6.5.3" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "eslint-config-airbnb-typescript > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-import@^2.22.1".
warning "eslint-config-airbnb-typescript > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-react@^7.21.5".
warning "eslint-config-airbnb-typescript > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-react-hooks@^4 || ^3 || ^2.3.0 || ^1.7.0".
warning "eslint-config-airbnb-typescript > eslint-config-airbnb-base@14.2.1" has unmet peer dependency "eslint-plugin-import@^2.22.1".
[4/4] Building fresh packages...
Done in 106.49s.       <===== After I ran  "rm -rf /root/.cache/prisma/, takes only 12 secs on my laptop
```
`yarn install` hangs until I manually remove /root/.cache/prisma/, which is pretty annoying. I wasted a days trying to find the problem until I found this issue.

Hope it gets resolved soon.

**Comment by guille-mas** [comment]
Using docker alpine-based image here, running yarn install and Prisma client generator inside the container. Adding ` binaryTargets = ["native"]` to my generator fixed the issue.


```
generator client {
  provider = "prisma-client-js"
  binaryTargets = ["native"]
}
```

**Comment by leomao10** [comment]
btw, I think this one is related to #7037, and #7222 should fix the issue for now.

**Comment by janpio** [comment]
Yes @leomao10, can everyone who experience this problem be follow these instructions and report back?

> Someone could try this integration release `prisma@2.24.0-integration-docker-workaround.1` for the CLI, based on https://github.com/prisma/prisma/pull/7221 that I quickly hacked together based on @leomao10 's comment: https://github.com/prisma/prisma/issues/7037#issuecomment-846370629 Let me know if this fixes the problem and we could merge this workaround into our next release to unblock all of you.

https://github.com/prisma/prisma/issues/7037#issuecomment-846392650

**Comment by JacobMuchow** [comment]
@janpio  The workaround release works for me! Upgraded docker desktop back up to 3.3.3. Thanks so much!

**Comment by janpio** [comment]
We have a real `dev` version of `2.24.0` out that should include this fix as well: `2.24.0-dev.34`. Please try that and report back if this solves the problem for you. It will most probably be included in our release tomorrow.

**Comment by desaiuditd** [comment]
`2.24.0-dev.34` did not work for me. I tried to do `yarn install` for a repo where `"@prisma/client": "^2.24.0-dev.34"` is included. I used `node:15.14.0-alpine3.13` docker image for this. And `yarn install` still continues to hang forever. I'm using Docker Desktop 3.3.3.

**Comment by janpio** [comment]
2.24.0 is out 👍

**Comment by janpio** [comment]
Can you try with `2.24.0` @desaiuditd? If this does not work, we have to keep this issue open or create a new one :( 

Can all the other people who participated in the issue maybe also try the new release and let us know if it works? Thanks!

**Comment by eddywashere** [comment]
2.24.0 worked for me. I did run into an issue but that was my fault. I should have done `yarn upgrade @prisma/client@2.24.0` vs updating package.json then running yarn, then removing my `yarn.lock` file and rerunning yarn 😬.

**Comment by JacobMuchow** [comment]
2.24.0 works for me as well!

**Comment by desaiuditd** [comment]
> Can you try with `2.24.0` @desaiuditd? If this does not work, we have to keep this issue open or create a new one :(
> 
> Can all the other people who participated in the issue maybe also try the new release and let us know if it works? Thanks!

All good. I also had to bump `prisma` package along with `@prisma/client`. It's working now.

**Comment by davi-mbatista** [comment]
With `@prisma/client` `v2.24.0`  I'm again experiencing `yarn install` hanging on docker at:
```
[5/5] Building fresh packages...
```
Anyone else facing the same?

**Comment by valerii15298** [comment]
Haveing the same problem with prisma, pnpm and docker. Cannot install prisma, it hangs indefinitely. Seems it because of prisma postinstall script.
https://github.com/pnpm/pnpm/discussions/4329

**Comment by Jolg42** [comment]
Thanks for mentioning that @valerii15298, I created an issue about it here https://github.com/prisma/prisma/issues/11791

Note: I will lock this issue, if you experience any problem related to this topic, please open a new issue or discussion 💚

**Comment by erlangparasu** [comment]
Setting `ENV PRISMA_SKIP_POSTINSTALL_GENERATE=true` in Dockerfile does not work for me

**Comment by prashantbhudwal** [comment]
Downgrade to docker 4.26
Disable Rosetta from docker settings
Build

Specs
Macos
M2
Sonoma

I have no clue why this works. But it does.


If you don't downgrade the docker, you will get a segmentation error from qemu.
